vignettes/tech-dm-keyed.Rmd

```diff
@@ -0,0 +1,216 @@
+---
+title: "Manipulating individual tables"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Technical: Manipulating individual tables}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+``````{r setup, include = FALSE}
+source("setup/setup.R")
+``````
+
+This vignette deals with situations where you want to transform tables of your `dm` and then update an existing table or add a new table to the `dm`.
+There are two straightforward approaches:
+
+1. individually subsetting the tables relevant to the calculation, perform the necessary transformations, add the result to the `dm` (or replace an existing table), and establish the key relations.
+1. do all this within the `dm` object by zooming to a table and manipulating it while maintaining the key relations whenever possible.
+
+We will explore the first approach here. For the second approach, see `vignette("tech-dm-zoom")`.
+
+## Enabling {dplyr}-workflow within a `dm`
+
+Subsetting a table of a `dm` object produces an object of class `dm_keyed_tbl`.
+
+{dm} provides methods for many of the {dplyr}-verbs for a `dm_keyed_tbl` which behave the way you are used to, affecting only the table and leaving the rest of the `dm` untouched.
+
+When you are finished with transforming the table, there are three options to proceed:
+
+1. use `dm()` if you want to replace the original tables with the new table
+1. use `new_dm()`/`dm()` if you want to create a brand new `dm` object 
+
+When employing these options, the resulting tables in the `dm` will have all the primary and foreign keys available that could be tracked from the original table.
+
+## Examples
+
+So much for the theory, but how does it look and feel?
+To explore this, we once more make use of our trusted {nycflights13} data.
+
+### Use case 1: Add a new column to an existing table
+
+Imagine you want to have a column in `flights`, specifying if a flight left before noon or after.
+Just like with {dplyr}, we can tackle this with `mutate()`.
+Let us do this step by step:
+
+```{r zoom}
+library(dm)
+library(dplyr)
+
+flights_dm <- dm_nycflights13()
+flights_dm
+
+flights_dm_tbl <-
+  flights_dm %>% 
+  dm_get_tables(keyed = TRUE)
+
+flights_tbl <- flights_dm_tbl$flights
```

Comment:
`flights_tbl` and `flights_dm_tbl` are too similar names.

`flights_dm_tbl` could be something like `dm_nycflights13_tibblified` or `dm_nycflights13_tbl` (maybe too close to `dm_nycflights13` :zany_face:)